airflow/jobs.py

```diff
@@ -550,46 +781,83 @@ def process_dag(self, dag, queue):
         self.logger.debug("Unlocking DAG (scheduler_lock)")
         db_dag = (
             session.query(DagModel)
-            .filter(DagModel.dag_id == dag.dag_id)
-            .first()
+                .filter(DagModel.dag_id == dag.dag_id)
+                .first()
         )
         db_dag.scheduler_lock = False
         session.merge(db_dag)
         session.commit()
 
         session.close()
 
-    @provide_session
-    def prioritize_queued(self, session, executor, dagbag):
-        # Prioritizing queued task instances
+    @staticmethod
+    def get_running_tasks(session, dag_id, task_ids):
```

Comment:
Yeah, I'll have to rebase to master in another PR after this one gets reviewed. Since this PR is fairly complicated, we want to deploy it on our clusters and we're running off of `airbnb_rb1.7.1_4`.